generateNeoForge.py

```diff
@@ -0,0 +1,209 @@
+from meta.common.neoforge import (
+    NEOFORGE_COMPONENT,
+    INSTALLER_MANIFEST_DIR,
+    VERSION_MANIFEST_DIR,
+    DERIVED_INDEX_FILE,
+    INSTALLER_INFO_DIR,
+    FORGEWRAPPER_MAVEN,
+)
+from meta.common.mojang import MINECRAFT_COMPONENT
+from meta.model import (
+    MetaVersion,
+    Dependency,
+    Library,
+    GradleSpecifier,
+    MojangLibraryDownloads,
+    MojangArtifact,
+    MetaPackage,
+)
+from meta.model.neoforge import (
+    NeoForgeVersion,
+    NeoForgeInstallerProfileV2,
+    InstallerInfo,
+    DerivedNeoForgeIndex,
+)
+from meta.model.mojang import MojangVersion
+
+LAUNCHER_DIR = launcher_path()
+UPSTREAM_DIR = upstream_path()
+STATIC_DIR = static_path()
+
+ensure_component_dir(NEOFORGE_COMPONENT)
+
+
+def eprint(*args, **kwargs):
+    print(*args, file=sys.stderr, **kwargs)
+
+
+def version_from_build_system_installer(
+    installer: MojangVersion,
+    profile: NeoForgeInstallerProfileV2,
+    version: NeoForgeVersion,
+) -> MetaVersion:
+    v = MetaVersion(name="NeoForge", version=version.rawVersion, uid=NEOFORGE_COMPONENT)
+    v.requires = [Dependency(uid=MINECRAFT_COMPONENT, equals=version.mc_version_sane)]
+    v.main_class = "io.github.zekerzhayard.forgewrapper.installer.Main"
+
+    # FIXME: Add the size and hash here
+    v.maven_files = []
+
+    # load the locally cached installer file info and use it to add the installer entry in the json
+    info = InstallerInfo.parse_file(
+        os.path.join(UPSTREAM_DIR, INSTALLER_INFO_DIR, f"{version.long_version}.json")
+    )
+    installer_lib = Library(
+        name=GradleSpecifier(
+            "net.neoforged", "forge", version.long_version, "installer"
+        )
+    )
+    installer_lib.downloads = MojangLibraryDownloads()
+    installer_lib.downloads.artifact = MojangArtifact(
+        url="https://maven.neoforged.net/%s" % (installer_lib.name.path()),
+        sha1=info.sha1hash,
+        size=info.size,
+    )
+    v.maven_files.append(installer_lib)
+
+    for upstream_lib in profile.libraries:
+        forge_lib = Library.parse_obj(upstream_lib.dict())
+        if forge_lib.name.is_log4j():
+            continue
+
+        if (
+            forge_lib.name.group == "net.neoforged"
+            and forge_lib.name.artifact == "forge"
+            and forge_lib.name.classifier == "universal"
+        ):
+            forge_lib.downloads.artifact.url = (
+                "https://maven.neoforged.net/%s" % forge_lib.name.path()
+            )
+        v.maven_files.append(forge_lib)
+
+    v.libraries = []
+
+    wrapper_lib = Library(
+        name=GradleSpecifier("io.github.zekerzhayard", "ForgeWrapper", "1.5.6-prism")
+    )
+    wrapper_lib.downloads = MojangLibraryDownloads()
+    wrapper_lib.downloads.artifact = MojangArtifact(
+        url=FORGEWRAPPER_MAVEN % (wrapper_lib.name.path()),
+        sha1="b059aa8c4d2508055c6ed2a2561923a5e670a5eb",
+        size=34860,
+    )
```

Comment:
We should extract these constants maybe

Reply:
probably :eyes:

Reply:
Please do :3

Reply:
I mean so we can have the same lib for forge and neoforge